src/psyclone/tests/gocean1p0_opencl_test.py

```diff
@@ -441,6 +592,105 @@ def test_psy_init_with_options(kernel_outputdir):
+    statement to launch OpenCL kernels. '''
+    api_config = Config.get().api_conf("gocean1.0")
+    monkeypatch.setattr(api_config, "_debug_mode", debug_mode)
+    psy, _ = get_invoke("single_invoke.f90", API, idx=0)
+    sched = psy.invokes.invoke_list[0].schedule
+    # Currently, moving the boundaries inside the kernel is a prerequisite
+    # for the GOcean gen_ocl() code generation.
+    trans = GOMoveIterationBoundariesInsideKernelTrans()
+    for kernel in sched.coded_kernels():
+        trans.apply(kernel)
+
+    otrans = OCLTrans()
+    otrans.apply(sched)
+    generated_code = str(psy.gen)
+
+    # Set the boundaries for this kernel
+    expected = '''
+        xstart = cu_fld%internal%xstart
+        xstop = cu_fld%internal%xstop
+        ystart = cu_fld%internal%ystart
+        ystop = cu_fld%internal%ystop'''
+
+    # Cast dl_esm_inf pointers to cl_mem handlers
+    expected += '''
+      cu_fld_cl_mem = transfer(cu_fld%device_ptr, cu_fld_cl_mem)
+      p_fld_cl_mem = transfer(p_fld%device_ptr, p_fld_cl_mem)
+      u_fld_cl_mem = transfer(u_fld%device_ptr, u_fld_cl_mem)'''
+
+    # Call the set_args subroutine with the boundaries corrected for the
+    # OpenCL 0-indexing
+    expected += '''
+      CALL compute_cu_code_set_args(kernel_compute_cu_code, \
+cu_fld_cl_mem, p_fld_cl_mem, u_fld_cl_mem, \
+xstart - 1, xstop - 1, \
+ystart - 1, ystop - 1)'''
+
+    # Set up globalsize and localsize values
+    expected += '''
+      globalsize = (/p_fld%grid%nx, p_fld%grid%ny/)
+      localsize = (/64, 1/)'''
+
+    if debug_mode:
+        # Check that the globalsize first dimension is a multiple of
+        # the localsize first dimension
+        expected += '''
+      IF (mod(p_fld%grid%nx, 64) .ne. 0) THEN
+        CALL check_status("Global size is not a multiple of local size \
+(mandatory in OpenCL < 2.0).", -1)
+      END IF'''
+
+    expected += '''
+      ! Launch the kernel'''
+
+    if debug_mode:
+        # Check that there is no pending error in the queue before launching
+        # the kernel
+        expected += '''
+      ierr = clFinish(cmd_queues(1))
+      CALL check_status('Errors before compute_cu_code launch', ierr)'''
+
+    expected += '''
+      ierr = clEnqueueNDRangeKernel(cmd_queues(1), kernel_compute_cu_code, \
+2, C_NULL_PTR, C_LOC(globalsize), C_LOC(localsize), 0, C_NULL_PTR, \
+C_NULL_PTR)
+      !'''
+
+    if debug_mode:
+        # Check that there are no errors during the kernel launch or during
+        # the execution of the kernel.
+        expected += '''
+      CALL check_status('compute_cu_code clEnqueueNDRangeKernel', ierr)
+      ierr = clFinish(cmd_queues(1))
+      CALL check_status('Errors during compute_cu_code', ierr)'''
+
+    assert expected in generated_code
+    assert GOcean1p0OpenCLBuild(kernel_outputdir).code_compiles(psy)
+
+
+def test_grid_proprty(kernel_outputdir):
+    # pylint: disable=unused-argument
+    ''' Test that using nx and ny from the gocean property dictionary
```

Comment:
I think this test can be removed as it is covered by the previous one?

Reply:
Agreed, removed the test.